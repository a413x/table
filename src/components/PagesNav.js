import React, { Component } from "react";
import PropTypes from 'prop-types';

import './styles/pages.css';

class PagesNav extends Component {
  constructor(props){
    super(props);
    this.nextPrevClick = this.nextPrevClick.bind(this);
    this.pageChange = this.pageChange.bind(this);
  }
  pageChange(e){
    const num = +e.target.value;
    if(num < 1 || num > this.props.pages) return;
    this.props.onPageChange(num);
  }
  nextPrevClick(direction){
    this.props.onPageChange(+this.props.currentPage + direction);
  }
  render(){
    const currPage = this.props.currentPage;
    const pages = this.props.pages;

    let prevEn = true;
    let nextEn = true;
    if(currPage == 1) prevEn = false;
    if(currPage == pages) nextEn = false;

    return (
      <div className='container d-flex'>
        <button className = 'btn btn-light' disabled = {prevEn ? null : true}
          onClick = {()=>this.nextPrevClick(-1)}>Назад</button>
        <nav className = 'd-flex w-100 justify-content-around'>
          <form className = 'form-inline'>
            <label>Страница</label>
            <input className="form-control m-2" id='pageInput' type="number"
              value={currPage} onChange={this.pageChange}/>
            <label>из {pages}</label>
          </form>
          <form className = 'form-inline'>
            <label className = 'mr-2'>Строк на странице:</label>
            <select onChange = {(e) => this.props.onNumberShownChange(+e.target.value)}>
              <option>5</option>
              <option>10</option>
              <option>50</option>
            </select>
          </form>
        </nav>
        <button className = 'btn btn-light' disabled = {nextEn ? null : true}
          onClick = {()=>this.nextPrevClick(1)}>Далее</button>
      </div>
    );
  }
}

PagesNav.propTypes = {
  pages: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  onNumberShownChange: PropTypes.func.isRequired
};

export default PagesNav;
