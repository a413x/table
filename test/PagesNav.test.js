import React from "react";
import {shallow,mount} from 'enzyme';

import PagesNav from "../src/components/PagesNav";

describe('tests for PagesNav component',() => {
  const mockonPageChange = jest.fn();
  const mockOnNumberShownChange = jest.fn();
  const props = {
    pages: 2,
    currentPage: 1,
    onPageChange: mockonPageChange,
    onNumberShownChange: mockOnNumberShownChange
  };

  const pagesNavContainer = shallow(<PagesNav {...props} />);

  it('renders properly',()=>{
    expect(pagesNavContainer).toMatchSnapshot();
  });

  it('pageChange called on input',()=>{
    const nextPage = 2;
    pagesNavContainer.find('#pageInput').simulate('change',{
      target:{value:nextPage}
    });
    expect(mockonPageChange).toHaveBeenCalledWith(nextPage);
  });

  it('pageChange called on button prev',()=>{
    pagesNavContainer.find('button').first().simulate('click');
    expect(mockonPageChange).toHaveBeenCalledWith(0);
  });

  it('pageChange called on button next',()=>{
    pagesNavContainer.find('button').last().simulate('click');
    expect(mockonPageChange).toHaveBeenCalledWith(2);
  });

  it('onNumberShownChange called on select change',()=>{
    pagesNavContainer.find('select').simulate('change',{
      target:{value:10}
    });
    expect(mockOnNumberShownChange).toHaveBeenCalledWith(10);
  });
});
