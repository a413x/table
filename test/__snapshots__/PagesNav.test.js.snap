// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for PagesNav component renders properly 1`] = `
<div
  className="container d-flex"
>
  <button
    className="btn btn-light"
    disabled={true}
    onClick={[Function]}
  >
    Назад
  </button>
  <nav
    className="d-flex w-100 justify-content-around"
  >
    <form
      className="form-inline"
    >
      <label>
        Страница
      </label>
      <input
        className="form-control m-2"
        id="pageInput"
        onChange={[Function]}
        type="number"
        value={1}
      />
      <label>
        из 
        2
      </label>
    </form>
    <form
      className="form-inline"
    >
      <label
        className="mr-2"
      >
        Строк на странице:
      </label>
      <select
        onChange={[Function]}
      >
        <option>
          5
        </option>
        <option>
          10
        </option>
        <option>
          50
        </option>
      </select>
    </form>
  </nav>
  <button
    className="btn btn-light"
    disabled={null}
    onClick={[Function]}
  >
    Далее
  </button>
</div>
`;
